frontend/app/mocks/subscription-api.server.ts

```diff
@@ -0,0 +1,91 @@
+import { HttpResponse, http } from 'msw';
+import { z } from 'zod';
+
+import { db } from './db';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('subscription-api.server');
+
+const subscriptionApiSchema = z.object({
+  id: z.string(),
+  sin: z.string(),
+  email: z.string(),
+  subscribed: z.boolean(),
+  preferredLanguage: z.string(),
+});
+
+/**
+ * Server-side MSW mocks for the subscription API.
+ */
+export function getSubscriptionApiMockHandlers() {
+  log.info('Initializing Subscription API mock handlers');
+
+  return [
+    //
+    // Handler for GET request to retrieve email alerts decription by sin
+    //
+    http.get('https://api.example.com/v1/subscriptions/:userId', ({ params, request }) => {
+      log.debug('Handling request for [%s]', request.url);
+
+      const url = new URL(request.url);
+      const alertType = url.searchParams.get('alertType');
+      const parsedUserId = z.string().safeParse(params.userId);
+
+      if (!parsedUserId.success || alertType === null) {
+        throw new HttpResponse(null, { status: 400 });
+      }
+
+      const subscriptionEntity = db.subscription.findFirst({
+        where: { sin: { equals: parsedUserId.data }, alertType: { equals: alertType } },
+      });
+
+      if (!subscriptionEntity) {
+        return HttpResponse.json(null, { status: 204 });
+      }
+
+      return HttpResponse.json({
+        id: subscriptionEntity.id,
+        sin: subscriptionEntity.sin,
+        email: subscriptionEntity.email,
+        subscribed: subscriptionEntity.subscribed,
+        preferredLanguage: subscriptionEntity.preferredLanguage,
+      });
+    }),
+
+    //
+    // Handler for PUT request to update email alerts decription
+    //
+    http.put('https://api.example.com/v1/subscriptions/update', async ({ request }) => {
```

Comment:
```suggestion
    http.put('https://api.example.com/v1/subscriptions', async ({ request }) => {
```

`PUT` is used for updating - no need to add `update` to the path.

Reply:
updated